lib/chef/resource/registry_key.rb

```diff
@@ -154,45 +152,33 @@ class RegistryKey < Chef::Resource
       # See lib/chef/resource_reporter.rb for more information.
       attr_reader :unscrubbed_values
 
-      def initialize(name, run_context = nil)
-        super
-        @values, @unscrubbed_values = [], []
-      end
+      VALID_VALUE_HASH_KEYS = %i{name type data}.freeze
 
       property :key, String, name_property: true
 
-      VALID_VALUE_HASH_KEYS = %i{name type data}.freeze
-
-      def values(arg = nil)
-        if not arg.nil?
-          if arg.is_a?(Hash)
-            @values = [ Mash.new(arg).symbolize_keys ]
-          elsif arg.is_a?(Array)
-            @values = []
-            arg.each do |value|
-              @values << Mash.new(value).symbolize_keys
-            end
+      property :values, [Hash, Array],
+        default: [],
+        coerce: proc { |v|
```

Comment:
Do we have specs that validate the data types we can pass and the coerce already?

Reply:
Yep we have some unit tests for the `values` property here. I tested against those while making the changes so that nothing breaks. https://github.com/chef/chef/blob/296edf56682fcf5c6f512bc8255325b869a4b36c/spec/unit/resource/registry_key_spec.rb#L79-L141

Reply:
unit tests on macOS are failing though for some reason. Will take a look now.